ls1intum/Artemis
# Enhancements of artemis date pipe

### Checklist

### Motivation and Context
The date formats introduced [in this PR](https://github.com/ls1intum/Artemis/pull/1488) still not completely comply with date / time formats.

### Description
- [x] Hours, minutes and seconds will be displayed with a leading zero
- [x] I've removed the *long-time* format. Time will always be displayed in the form of `09:05:02`.
        **Justification:** the current format (`3:45 PM`) might be confusing to non-native english speakers (`AM` vs. `PM`) and it does not provide any advantage over the new format (`15:45`).
- [x] I've tested that for an undefined / null date object always an empty string will be returned.
- [x] I've adjusted the date format on the course overview page.
- [x] I've changed the short date format to `2019-03-18` (`en` locale) and `18.03.2019` (`de` locale)

**_format_**|**de**|**en**
:-----:|:-----:|:-----:
**short**|17.12.2019 15:03|2019-12-17 15:03
**long**|17. Dez. 2019 15:03|Dec 17, 2019 15:03
**short-date**|17.12.2019|2019-12-17
**long-date**|17. Dez. 2019|Dec 17, 2019
**time**|15:03|15:03

### Steps for Testing
Simply log in and inspect places where dates are shown. Especially in the course overview (see Screenshot).

### Screenshot

![image](https://user-images.githubusercontent.com/15168372/83547838-18eea780-a503-11ea-9a09-1db8931bf692.png)
